src/main/scala/fi/oph/koski/validation/KoskiValidator.scala

```diff
@@ -702,6 +703,115 @@ class KoskiValidator(tutkintoRepository: TutkintoRepository, val koodistoPalvelu
     }
   }
 
+  // TOR-982 - Ammatillisen koulutuksen yhteisten osien laajuuksien yms validaatioita
+  private val yhteiset = List("101053", "101054", "101055", "400012", "400013", "400014")
+  private def validateYhteisetOsat(suoritus: Suoritus): HttpStatus = {
+    def validateEiSamojaKoodeja(suoritus: Suoritus): HttpStatus = suoritus match {
+      case a: AmmatillisenTutkinnonSuoritus if a.valmis => {
+        val yhteistenOsasuoritustenKoodit = a.osasuoritukset.toList.flatten.filter(o => o.arvioitu && yhteiset.contains(o.koulutusmoduuli.tunniste.koodiarvo)).map(_.koulutusmoduuli.tunniste.koodiarvo)
+        HttpStatus.validate(
+          yhteistenOsasuoritustenKoodit.distinct.size == yhteistenOsasuoritustenKoodit.size
+        )(KoskiErrorCategory.badRequest.validation.rakenne.duplikaattiOsasuoritus(
+          s"Suorituksella ${suorituksenTunniste(suoritus)} on useampi yhteinen osasuoritus samalla koodilla"))
+      }
+      case _ => HttpStatus.ok
+    }
+
+    def validateYhteislaajuus(suoritus: Suoritus): HttpStatus = suoritus match {
+      case a: AmmatillisenTutkinnonSuoritus if a.valmis && a.suoritustapa.koodiarvo == "reformi" => {
+
+        val yhteislaajuus = a.osasuoritukset.toList.flatten.filter(o => o.arvioitu && yhteiset.contains(o.koulutusmoduuli.tunniste.koodiarvo))
+          .foldLeft(0.0) { (a, b) => a + (b.koulutusmoduuli.laajuus match {
```

Comment:
Parantuisiko tässä luettavuus jos käyttäisi `foldLeft`:in sijaan `map`:ia, `flatMap`:ia tai `collect`:ia sekä `sum`-aggregaatiota?